pulumi/infra/network.py

```diff
@@ -0,0 +1,139 @@
+import ipaddress
+import math
+from typing import Optional
+
+import pulumi_aws as aws
+from infra.config import import_aware_opts
+
+import pulumi
+
+# TODO: Add policy requiring that each networking-related
+# infrastructure component have a "Name" (note capitalization) tag,
+# since this is how these objects get their names. Otherwise, they
+# have no name, which makes interaction via the AWS UI slightly more
+# awkward than it needs to be.
+#
+# Security groups don't seem to work this way, though. They get a name
+# independent of a Name tag (though the AWS UI shows both, somewhat
+# confusingly).
+
+
+class Network(pulumi.ComponentResource):
```

Comment:
TBH I don't have much background on this sort of stuff, in particular cidr. may want second pair of eyes